src/lib.rs

```diff
@@ -104,86 +105,127 @@ mod parser;
 const VERSION: &str = "0.0.1";
 const LANG: &str = "rust";
 
-impl Options {
-    /// Add a name option for the unconnected connection.
+#[doc(hidden)]
+pub mod options_typestate {
+    /// `Options` typestate indicating
+    /// that there has not yet been
+    /// any auth-related configuration
+    /// provided yet.
+    #[derive(Debug, Copy, Clone, Default)]
+    pub struct Unauthenticated;
```

Comment:
Don't have a better recommendation but not a huge fan of this. Plain?

Reply:
NoAuth?

Reply:
switched to NoAuth

Reply:
ok thanks.